reactor-core/src/test/java/reactor/core/publisher/FluxSwitchOnFirstTest.java

```diff
@@ -98,6 +101,110 @@ public void shouldNotSubscribeTwiceConditional() {
                   .hasMessage("FluxSwitchOnFirst allows only one Subscriber");
     }
 
+    @Test
+    public void shouldNotSubscribeTwiceWhenCanceled() {
+        Schedulers.parallel().start();
+        Schedulers.single().start();
+        StepVerifier.create(Flux.just(1L)
+                                .doOnComplete(() -> {
+                                    try {
+                                        Thread.sleep(200);
+                                    }
+                                    catch (InterruptedException e) {
+                                        e.printStackTrace();
+                                    }
+                                })
+                                .hide()
+                                .publishOn(Schedulers.parallel())
+                                .cancelOn(new Scheduler() {
+                                    @Override
+                                    public Disposable schedule(Runnable task) {
+                                        return Disposables.composite();
+                                    }
+
+                                    @Override
+                                    public Worker createWorker() {
+                                        return new Worker() {
+                                            @Override
+                                            public Disposable schedule(Runnable task) {
+                                                return Disposables.composite();
+                                            }
+
+                                            @Override
+                                            public void dispose() {
+
+                                            }
+                                        };
+                                    }
+                                })
+                                .switchOnFirst((s, f) -> f)
+                                .doOnSubscribe(s ->
+                                    Schedulers
+                                        .single()
+                                        .schedulePeriodically(s::cancel, 10, 0, TimeUnit.MILLISECONDS)
+                                )
+        )
+                    .expectSubscription()
+                    .expectNext(1L)
+                    .expectNoEvent(Duration.ofSeconds(1))
+                    .thenCancel()
+                    .verifyThenAssertThat()
+                    .hasNotDroppedErrors();
+    }
+
+    @Test
+    public void shouldNotSubscribeTwiceConditionalWhenCanceled() {
+        Schedulers.parallel().start();
+        Schedulers.single().start();
+        StepVerifier.create(Flux.just(1L)
+                                .doOnComplete(() -> {
+                                    try {
+                                        Thread.sleep(200);
```

Comment:
is it possible to use smaller delays? It adds a constant time to the tests execution time, and we should keep it low

Reply:
done